app/controllers/components/course/controller_component_host.rb

```diff
@@ -52,6 +53,20 @@ def gamified?
       def can_be_disabled?
         true
       end
+
+      # Returns a model which the current component can use to interface with its persisted
+      # settings.
+      #
+      # @return [Class] The settings interface class
+      # @return [nil] if the class does not exist
+      def settings_class
+        @settings_class ||= "Course::Settings::#{name.demodulize}".safe_constantize
```

Comment:
This abstraction introduces some implicit assumptions. 

If I'm not wrong, one has to define the component in the controller component host (`Course::FooComponent`), which then has to be named similarly to the unique settings key (`settings_foo_component`) and settings of the component (`Course::Settings::Foo`). 

I have a 2 questions:
1. Is this a little too complex, and are there easier ways to simplify it? 
2. If we keep this, since it is widely utilised, it possible to document this somewhere, and guard this with tests?